cinema/permissions.py

```diff
@@ -0,0 +1,14 @@
+from rest_framework.permissions import BasePermission, SAFE_METHODS
+
+
+class IsAdminOrIfAuthenticatedReadOnly(BasePermission):
+
+    def has_permission(self, request, view):
+        return bool(
+            request.method in SAFE_METHODS
+            and request.user
+            and request.user.is_authenticated
+        ) or bool(
+            request.user
+            and request.user.is_staff
+        )

```

Comment:
2 suggestions:
1. No need to check request.user because request always has user
2. No need to convert everything to boolean using bool(). All your conditions return boolean, thus the result will be always boolean